### DISCUSSION TITLE
v4.0.42 pre-release

### DISCUSSION THREAD

**Discussion by autoSteve** [BJReplay/ha-solcast-solar#112] [General]
Enhancements, fixes and a defeat.

* Enhancement: For an initial installation, if you're unlucky then the Solcast API might be busy, and three retries will likely not be enough. So the integration now fails to load entirely in this situation, passing control back to Home Assistant to have it repeatedly attempt the startup until the sites can be retrieved.
* Enhancement: Minor spline bounces for the "moment" sensors could be observed at the beginning and end of each day (e.g. Power Now/Power +30 mins/Power +1 hour). These bounces are suppressed.
* Fix: A timer race could occur at midnight, causing sensors like "remaining for the day" to show an incorrect value for five minutes.
* Enhancement: Additional explanations have been added to the readme (like for the "hard limit" option).
* Concedement of defeat: Applying a dampening factor, or the hard limit to site breakdown attributes made little sense, so these calculations have been removed. The only way to do it properly would be to have dampening factors broken down by site, which would get unweildy. (An example: One site may be shaded in the afternoon while the other is not.)

As always, both positive and negative feedback is most welcome, and this will contribute to getting the release ultimately published as latest. Report back, and don't be shy!

Feedback in the past has been gratefully received from @dtswk, @EddieSpaghettie, @ffm777, @gcoan, @hwb45, @iainfogg, @isorin, @jeffb1974, @kernow, @ladaan, @Markyt77, @nilogax, @Noodleyman, @ProphetOfDoom, @rholligan, @Unlik and @wimdebruyn, and continued thanks to them for their interest in beta releases (and alphas!) and for their contributions.

> ~~[!NOTE]~~
> ~~The BJReplay fork is **_still_** not published for searching in HACS, so continued use of a manual install in HACS is the go. As explained earlier we're going to keep any releases as pre-release on the fork until that happens, as any 'latest' version change would require modification of that HACS pull request.~~
(Strike-through spoiler...)

## What's Changed
* Initial sites load fail reporting and fall-back on HA auto-retry startup by @autoSteve
* Suppress spline bounce in moment splines by @autoSteve
* Recalculate splines at midnight before sensors update by @autoSteve
* Readme updates by @autoSteve
* Dampening and hard limit removed from per-site forecast breakdowns (too hard, too misleading) by @autoSteve

**Full Changelog**: https://github.com/BJReplay/ha-solcast-solar/compare/v4.0.41...v4.0.42

**Comment by iainfogg** [comment]
Just installed, thanks, I'll shout if there are any issues.

**Comment by autoSteve** [comment]
I know there have been quite a few betas released of late, and that "upgrade fatigue" can set in. But each new release takes us one step closer to "latest", which will be unleashed on thousands. It... Is... Soon.

Sooner than you think.

As much as I liked the release number v4.0.42. Because 42 is _the answer_ as defined by Douglas Adams in Hitchhikers, v4.0.43 pre-release is out.

It is not earth-shattering, if I'm honest. Well, it might be if you had Home Assistant stick its toes in the air for days while you were out and about for weeks, only to come back and resurrect it and not have Solcast forecasts automagically retrieved on startup... #110.

Although this particular enhancement will impact hardly anyone, **an upgrade will be highly valued by @BJReplay and I** to ensure that the code runs cleanly for as many as possible. For why? Watch this space... Upgrade now.

Thanks, @gcoan. One less rake handle for folks stumbling about to get hit in the face with, when they inevitably step on that rake head after an extended HA failure...

There will not be a new v4.0.43 discussion topic. The kind of failure that causes the issues that this release addresses does not warrant separate discussion. Equally, this v4.0.42 discussion has not exactly been running hot, despite 110 downloads with zero squeaks. And zero issues apart from @gcoan's enhancement request. **Let's hear some more, "It's fine"**...

**Reply by gcoan** [reply]
Did you see my comment on the PR for this https://github.com/BJReplay/ha-solcast-solar/pull/114#pullrequestreview-2227516172

Happy if you tell me that I know nothing and shouldn't worry myself

**Reply by autoSteve** [reply]
I did not. An earlier branch was deleted, and a new one created. Tell me that something did not go 'boom'...

**Reply by autoSteve** [reply]
That link shows no comment either...

**Reply by autoSteve** [reply]
I saw that you had reviewed, and GitHub says so, but the review is just not there... Gone.

<img width="490" alt="image" src="https://github.com/user-attachments/assets/a57836f9-1616-4d0d-97b9-008db7b3c95a">

**Reply by gcoan** [reply]
Weird ***t

I can see the comment still, here it is:

![image](https://github.com/user-attachments/assets/834c967c-5d46-48e1-89d8-e8243fa6cbd9)

**Reply by autoSteve** [reply]
That has been like it for quite some time. Meaningful data could be pulled from a firing exception, so it's just an unused variable. Cheers.

**Reply by BJReplay** [reply]
Nice. 38 installs so far.

**Comment by gcoan** [comment]
Installed 0.43, all looking fine, no issues or warnings in the log.  Got a solcast sites retrieval error 429 on startup and it seamlessly used the cache

**Comment by wimdebruyn** [comment]
0.43 installed and no issues

**Comment by Nilogax** [comment]
0.43 here, nothing to report.  File under "Mostly Harmless"

**Reply by BJReplay** [reply]
> Mostly Harmless

We really wanted 42 to be the "Release Release" for obvious reason.

**Comment by BJReplay** [comment]
P.S. Any feedback welcomed on https://github.com/BJReplay/ha-solcast-solar/tree/v4.1 - mainly updates to the readme and change are https://github.com/BJReplay/ha-solcast-solar/compare/v4.0.43...v4.1
